File: FoosballGames.Contracts/GetFoosballGames.cs
using System.Collections.Generic;
using FoosballGames.Infrastructure.Messaging;

namespace FoosballGames.Contracts
{
    public record GetFoosballGames : IQuery<FoosballGamesResponse>
    {
    }

    public record FoosballGamesResponse(IReadOnlyCollection<FoosballGame> FoosballGames);
}

File: FoosballGames.Contracts/GetFoosballGames.fs
namespace FoosballGames.Contracts

open System.Collections.Generic
open FoosballGames.Contracts
open FoosballGames.Infrastructure.Messaging

type FoosballGamesResponse = {FoosballGames: FoosballGame IReadOnlyCollection}

// Record can't be empty, so we're using a normal class
type GetFoosballGames() =
    interface IQuery<FoosballGamesResponse>

